packages/rollup-plugin-workbox/src/index.ts

```diff
@@ -1,15 +1,23 @@
 import {
   generateSW as _generateSw,
   injectManifest as _injectManifest,
-  GenerateSWConfig,
-  InjectManifestConfig,
+  GenerateSWOptions,
+  InjectManifestOptions,
 } from 'workbox-build';
 import * as prettyBytes from 'pretty-bytes';
 import * as rollup from 'rollup';
 import replace from '@rollup/plugin-replace';
 import { terser } from 'rollup-plugin-terser';
 import resolve from '@rollup/plugin-node-resolve';
 
+interface RollupPartial {
+  mode?: string | null;
+}
+
+type RollupGenerateSWOptions = GenerateSWOptions & RollupPartial;
+
+type RollupInjectManifestOptions = InjectManifestOptions & RollupPartial;
+
```

Comment:
It feels weird to need to construct types here. Are they not included in the new version or an updated types packages?

Reply:
The new version includes following types:

- `GenerateSWOptions`
- `WebpackGenerateSWOptions`
- `InjectManifestOptions`
- `WebpackInjectManifestOptions`

The problem comes from `InjectManifestOptions` type which does not support `mode` property.
At the same time, `WebpackInjectManifestOptions` does support `mode`, but not `globDirectory`.

So I decided to create types for both, even though I could use `GenerateSWOptions` as is.
This is done for consistency and to ensure we can rely on `mode` also in `generateSW`.

If you think types need to be in a separate file `types.ts`, let's move them there.

Reply:
We support the `mode` property for `injectManifest`, which workbox itself doesnt support, so to me there it makes sense. It shouldnt be needed for `generateSw` though, because workbox should support the `mode` property there already

Reply:
Thanks, indeed `GenerateSWOptions` contains this property. Updated to only add one type.